src/components/core/DynamicImage/DynamicImage.tsx

```diff
@@ -10,26 +17,34 @@ import {
   ImageSourcePropType,
   ImageStyle,
   ImageURISource,
+  LayoutChangeEvent,
   StyleProp,
   StyleSheet,
-  View
+  View,
+  ViewStyle
 } from 'react-native'
 
 import { ImageSkeleton } from 'app/components/image-skeleton'
 
 export type DynamicImageProps = {
   // Image source
-  image?: ImageSourcePropType
-  // Style to apply to the image itself
-  style?: StyleProp<ImageStyle>
+  source?: ImageSourcePropType
+  styles?: {
+    root?: ViewStyle
+    imageContainer?: ViewStyle
+    image?: StyleProp<ImageStyle>
+  }
+  style?: ViewStyle
```

Comment:
Should we maybe just get rid of the `style` prop? It might create confusion, what are your thoughts?

Reply:
yeah good q, i typically leave it cause its quite common to just want to update the root element, like adding padding, and it feels a bit over-engineered to always use the styles prop. thats just my experience, using className vs classes prop on the web. what do you think?

Reply:
I agree with that, my only worry is that it's a bit of boiler plate having to define both props on components. Maybe we make a helper type that encapsulates that? Something like: 

```ts
type Style = StyleProp<ViewStyle | ImageStyle | TextStyle>

type PropsWithStyles<
  Props extends Record<string, any>,
  Styles extends { root: Style } & Record<string, Style>
> = Props & {
  style: Styles['root']
  styles: Styles
}

export type DynamicImageProps = PropsWithStyles<
  {
    // Image source
    image?: ImageSourcePropType
    // Whether or not to immediately animate
    immediate?: boolean
    // Whether or not to use the default placeholder
    usePlaceholder?: boolean
  },
  {
    root: StyleProp<ImageStyle>
  }
>
```

Reply:
ohh wow love this! adding